src/main/java/com/marklogic/kafka/connect/source/KeyGenerator.java

```diff
@@ -0,0 +1,15 @@
+package com.marklogic.kafka.connect.source;
+
+import java.util.Map;
+
+public interface KeyGenerator {
+    String getUniqueKey();
```

Comment:
"unique" is potentially misleading here - we're not offering any guarantee that the value will be unique. Just do `generateKey` so that we don't imply something that may not be true (for example, if we support column-based keys, it's on the user to determine if the keys will be unique or not - and they may intentionally not want them to be unique).